### ISSUE TITLE
valueLabelTrue and valueLabelFalse props of <BooleanField> have no effect

### ISSUE THREAD

**Issue by MoonSeeing** [marmelab/react-admin#4047]
**What you were expecting:**

> If you need to override it, you can use the valueLabelTrue and valueLabelFalse props which both accept a string. Those strings may be translation keys:
```
<BooleanField source="published" valueLabelTrue="Has been published" valueLabelFalse="Has not been published yet" />
```
i want to override the display for bool value ,but it did  not on effect.

**Steps to reproduce:**
 ```
               <List>
                <Datagrid rowClick={rowClick} expand={PostPanel} optimized>
                   <BooleanField
                        source="commentable"
                        label="resources.posts.fields.commentable_short"
                        sortable={false}
                        valueLabelTrue="Has been published" 
                        valueLabelFalse="Has not been published yet"
                    />
                </Datagrid>
       </List>
``` in simple  post list

**Environment**

* React-admin version:3.0

**Comment by fzaninotto** [comment]
The doc says: 

> The BooleanField also includes an hidden text for accessibility (or to query in end to end tests). By default, it includes the translated label and the translated value, for example Published: false.

The `valueLabelTrue` and `valueLabelFalse` props do work, you just can't see the related text on a normal web browser. Only screen readers will say it. 

That being said, we can do better and provide both a fallback for screen readers AND a tooltip for most users. I'm marking this as an enhancement.

**Comment by MoonSeeing** [comment]
OK, got it

**Comment by djhi** [comment]
Closed by #4054
